plugins/inputs/logparser/grok/grok.go

```diff
@@ -281,7 +278,7 @@ func (p *Parser) ParseLine(line string) (telegraf.Metric, error) {
 		}
 	}
 
-	return metric.New(p.Measurement, tags, fields, p.tsModder.tsMod(timestamp))
```

Comment:
This line is the reason for all the grok and logparser changes.
`metric.New()` will return error if there are no fields present. In the journalparser plugin this may happen as the user may have configured the plugin to parse multiple journal fields. The journalparser plugin calls `ParseLine` for each journal field and then merges all the results together. And so one of the calls might not extract any fields.